### PR TITLE
Add iscsi into boot and update existin partitioning

### PR DESCRIPTION
**Pull request by dzedro**
http://10.100.98.90/tests/1343

### CODE REVIEW

File: tests/installation/partitioning_iscsi.pm [hunk 1]
```diff
@@ -13,14 +13,21 @@ use testapi;
...
-    send_key "alt-n";    # next
-    assert_screen "edit-proposal-settings";
+    if (check_screen("preparing-disk-use-entire-disk-button", 10)) {
+        send_key "alt-e";    # use entire iscsi disk
+        send_key "alt-d";    # edit proposal settings
+    }
+    else {
+        send_key "alt-e";    # edit proposal settings
+    }
+    assert_screen "preparing-disk-proposal-settings";
```
Comment (okurz) [inline tests/installation/partitioning_iscsi.pm]: would you mind combining this assert_screen with the check_screen in line 20 as in
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/f9aaf1b4c8f7f2e977f8d1b5b0cbbc726bd89255/tests/x11/sshxterm.pm#L19 and following?

**Comment by okurz** [comment]
one infile comment, looks good otherwise

**Comment by dzedro** [comment]
one infile comment ?

**Comment by okurz** [comment]
yes, the one you attended already. I just wanted to add a cover message to tell that I reviewed everything and did not find anything besides the one location which I commented.

looks ok, why WIP now?

File: tests/installation/partitioning_iscsi.pm (hunk 1 above)
Comment (dzedro) [inline tests/installation/partitioning_iscsi.pm]: doesn't work http://10.100.98.90/tests/1349/modules/partitioning_iscsi/steps/4

**Comment by dzedro** [comment]
Found one problem, have to fix it first, it will possibly affect this and another one PR

**Comment by dzedro** [comment]
updated

File: tests/installation/partitioning_iscsi.pm [hunk 2]
```diff
@@ -13,14 +13,15 @@ use testapi;
...
     send_key "alt-1";    # select ISCSI disk
     send_key "alt-n";    # next
-    assert_screen "preparing-disk-install-on";
-    send_key "alt-e";    # use entire iscsi disk
-    send_key "alt-n";    # next
-    assert_screen "edit-proposal-settings";
+    if (check_screen "preparing-disk-use-entire-disk-button", 10) {
+        send_key "alt-e";    # use entire iscsi disk
+    }
+    assert_screen "preparing-disk-overview";
```
Comment (okurz) [inline tests/installation/partitioning_iscsi.pm]: can these four lines be replaced by
```perl
assert_screen([qw/preparing-disk-overview preparing-disk-use-entire-disk-button/]);
if (match_has_tag 'preparing-disk-use-entire-disk-button') {
    send_key "alt-e";    # use entire iscsi disk
    assert_screen "preparing-disk-overview";
}
```

Comment (dzedro) [inline tests/installation/partitioning_iscsi.pm]: yes they can but what would be the benefit of this replace ?

Comment (okurz) [inline tests/installation/partitioning_iscsi.pm]: 1. on performance problems the check_screen with timeout of 10 seconds might fail, e.g. miss the screen
2. if the screen is not there we waste 10 seconds, not much, right

I won't mind merging this PR unchanged if you don't want to change

Comment (dzedro) [inline tests/installation/partitioning_iscsi.pm]: I will change it, it's no problem,I just wanted to know the reason :)

Comment (dzedro) [inline tests/installation/partitioning_iscsi.pm]: both needles can match http://10.100.98.90/tests/1519/modules/partitioning_iscsi/steps/2

Comment (sysrich) [inline tests/installation/partitioning_iscsi.pm]: hmmm..fix then needles, or change the logic back to how you had it? :)

**Pull request merged**